demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2273,15 +2284,64 @@ export class DemoMeetingApp
     this.log(`session stopped from ${JSON.stringify(sessionStatus)}`);
     this.log(`resetting stats in WebRTCStatsCollector`);
     this.statsCollector.resetStats();
+
+    const returnToStart = () => {
+      // If you want to make this a repeatable SPA, change this function to
+      // ```
+      // this.switchToFlow('flow-authenticate');
+      // ```
+      // and fix some state (e.g., video buttons).
+      //
+      // For simplicity, let's just reload the page.
+      window.location.href = window.location.pathname;
+    };
+
+    /**
+     * This is approximately the inverse of the initialization method above.
+     * This work only needs to be done if you want to continue using the page; if
+     * your app navigates away or closes the tab when done, you can let the browser
+     * clean up.
+     */
+    const onLeftMeeting = async () => {
+      // Clean up the timers for this.
+      this.audioVideo.unsubscribeFromActiveSpeakerDetector(this.activeSpeakerHandler);
+
+      // Stop listening to attendee presence.
+      this.audioVideo.realtimeUnsubscribeToAttendeeIdPresence(this.attendeeIdPresenceHandler);
+
+      // Stop watching device changes in the UI.
+      this.audioVideo.removeDeviceChangeObserver(this);
+
+      // Drop the audio input and output.
+      await this.audioVideo.chooseAudioInputDevice(null);
+      await this.audioVideo.chooseAudioOutputDevice(null);
+      this.audioVideo.unbindAudioElement();
+
+      // Clean up the loggers so they don't keep their `onload` listeners around.
+      setTimeout(async () => {
+        await this.meetingEventPOSTLogger?.dispose();
+        await this.meetingSessionPOSTLogger?.dispose();
```

Comment:
Doesn't this also call device controller dispose method which also dispose the audio io and video input? So do we need the code above?

Reply:
Why would you think it does that?

Reply:
oops. I misread this to meetingSession.dispose. Should we call meetingSession.dispose then? https://github.com/aws/amazon-chime-sdk-js/pull/1126/commits/8ab1cce7ffe35d8de673463a17715db4ee234577?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.ts#diff-bade3a8271704aa5eaa7df2a595d9aa8b18f8ebb3e1aa1fdf65f443c3abbdd37R114
You introduced several dispose but i did not see we use it here.

Reply:
My aim was to make it so existing apps that unregister in all the right places wouldn't also need to call dispose — the first draft of this papered over the problem by allowing a force cleanup, but avoiding leaks without that is better.